cockatrice/src/main.cpp

```diff
@@ -97,6 +100,19 @@ bool settingsValid()
         !settingsCache->getPicsPath().isEmpty();
 }
 
+void generateClientID()
```

Comment:
In a unit test that calls this method and nothing else, the test now depends on the global shared settings interface. Since this is global mutable state the test then has to remove the setting after running itself. Contrast with a version that returns QString without touching settings - we just call the method and check the return value. I know we don't have automated testing in place yet but it's one of my higher-priority goals for the project and I'd like to make the inclusion as smooth as I can.